modules/trusted-firmware-m/tfm_boards/common/nrf_provisioning.c

```diff
@@ -16,6 +16,51 @@
 #include "nrf_provisioning.h"
 #include <identity_key.h>
 #include <tfm_spm_log.h>
+#ifdef NRF53_SERIES
+#include <dfu/pcd_common.h>
+#include <spu.h>
+#include <hal/nrf_reset.h>
+
+static struct pcd_cmd *cmd = (struct pcd_cmd *)PCD_CMD_ADDRESS;
+
+static void pcd_write_cmd_lock_debug(void)
+{
+	cmd->magic = PCD_CMD_MAGIC_LOCK_DEBUG;
+	cmd->data = NULL;
+	cmd->len = 0;
+	cmd->offset = 0;
+}
+
+static bool pcd_read_cmd_done(void)
+{
+	return cmd->magic == PCD_CMD_MAGIC_DONE;
+}
+
+static enum tfm_plat_err_t disable_netcore_debug(void)
+{
+	/* NRF_RESET to secure. It will be configured non-secure after the provisioning is done. */
+	spu_peripheral_config_secure(NRF_RESET_S_BASE, SPU_LOCK_CONF_UNLOCKED);
+
+	/* Ensure that the network core is stopped. */
+	nrf_reset_network_force_off(NRF_RESET, true);
+
+	/* Debug lock command will be read in b0n startup. */
+	pcd_write_cmd_lock_debug();
+
+	/* Start the network core. */
+	nrf_reset_network_force_off(NRF_RESET, false);
+
+	/* Wait for the network core to start up. */
+	NRFX_DELAY_US(1000000);
+
+	if (!pcd_read_cmd_done()) {
+		SPMLOG_ERRMSG("Failed to lock debug in network core.");
+		return TFM_PLAT_ERR_SYSTEM_ERR;
+	}
```

Comment:
Should this be a busy-loop with some maximum wait time? Do we certainly know the exact time it takes for network core to write the struct content?

Reply:
There is a looping [mechanism](https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/pcd/src/pcd.c#L210-L217) in pcd.c with a relevant comment:

```
/* Wait for 1 second to avoid issue where network core
 * is unable to write to shared RAM.
 */
```
I think that setting Approtect should be rather fast compared to network core update, so likely 1 second should be enough. But we can add some looping to this.